src/Microsoft.NetCore.Analyzers/CSharp/Performance/CSharpDoNotUseCountAsyncWhenAnyAsyncCanBeUsed.Fixer.cs

```diff
@@ -160,14 +174,34 @@ private static void GetFixerForGreaterThanOrEqualExpression(BinaryExpressionSynt
             }
         }
 
-        private static void GetFixerForBinaryExpression(InvocationExpressionSyntax invocationExpression, LiteralExpressionSyntax literalExpression, int value, out SyntaxNode expression, out IEnumerable<SyntaxNode> arguments, out bool negate)
+        private static void GetFixerForBinaryExpression(ExpressionSyntax sourceExpression, bool isAsync, LiteralExpressionSyntax literalExpression, int value, out SyntaxNode expression, out IEnumerable<SyntaxNode> arguments, out bool negate)
         {
-            GetExpressionAndInvocationArguments(invocationExpression, out expression, out arguments);
+            GetExpressionAndInvocationArguments(sourceExpression, isAsync, out expression, out arguments);
             negate = (int)literalExpression.Token.Value == value;
         }
 
-        private static void GetExpressionAndInvocationArguments(InvocationExpressionSyntax invocationExpression, out SyntaxNode expression, out IEnumerable<SyntaxNode> arguments)
+        private static void GetExpressionAndInvocationArguments(ExpressionSyntax sourceExpression, bool isAsync, out SyntaxNode expression, out IEnumerable<SyntaxNode> arguments)
         {
+            while (sourceExpression is ParenthesizedExpressionSyntax parenthesizedExpression)
+            {
+                sourceExpression = parenthesizedExpression.Expression;
+            }
+
+            if (isAsync)
+            {
+                if (sourceExpression is AwaitExpressionSyntax awaitExpressionSyntax)
+                {
+                    sourceExpression = awaitExpressionSyntax.Expression;
```

Comment:
Can the source expression be something other then an `InvocationExpressionSyntax` here, say for error cases? If so, should we be more defensive and check `if (sourceExpression is AwaitExpressionSyntax awaitExpressionSyntax && awaitExpressionSyntax.Expression is InvocationExpressionSyntax)`?